lib/std/os/linux/x86.zig

```diff
@@ -400,75 +403,77 @@ noinline fn getContextReturnAddress() usize {
-          [link_offset] "p" (@offsetOf(ucontext_t, "link")),
-          [edi_offset] "p" (comptime gpRegisterOffset(REG.EDI)),
-          [esi_offset] "p" (comptime gpRegisterOffset(REG.ESI)),
-          [ebp_offset] "p" (comptime gpRegisterOffset(REG.EBP)),
-          [esp_offset] "p" (comptime gpRegisterOffset(REG.ESP)),
-          [ebx_offset] "p" (comptime gpRegisterOffset(REG.EBX)),
-          [edx_offset] "p" (comptime gpRegisterOffset(REG.EDX)),
-          [ecx_offset] "p" (comptime gpRegisterOffset(REG.ECX)),
-          [eax_offset] "p" (comptime gpRegisterOffset(REG.EAX)),
-          [eip_offset] "p" (comptime gpRegisterOffset(REG.EIP)),
-          [fs_offset] "p" (comptime gpRegisterOffset(REG.FS)),
-          [fpregs_offset] "p" (@offsetOf(ucontext_t, "mcontext") + @offsetOf(mcontext_t, "fpregs")),
-          [regspace_offset] "p" (@offsetOf(ucontext_t, "regspace")),
+        : [flags_offset] "i" (@offsetOf(ucontext_t, "flags")),
+          [link_offset] "i" (@offsetOf(ucontext_t, "link")),
+          [edi_offset] "i" (comptime gpRegisterOffset(REG.EDI)),
+          [esi_offset] "i" (comptime gpRegisterOffset(REG.ESI)),
+          [ebp_offset] "i" (comptime gpRegisterOffset(REG.EBP)),
+          [esp_offset] "i" (comptime gpRegisterOffset(REG.ESP)),
+          [ebx_offset] "i" (comptime gpRegisterOffset(REG.EBX)),
+          [edx_offset] "i" (comptime gpRegisterOffset(REG.EDX)),
+          [ecx_offset] "i" (comptime gpRegisterOffset(REG.ECX)),
+          [eax_offset] "i" (comptime gpRegisterOffset(REG.EAX)),
+          [eip_offset] "i" (comptime gpRegisterOffset(REG.EIP)),
+          [fs_offset] "i" (comptime gpRegisterOffset(REG.FS)),
+          [fpregs_offset] "i" (@offsetOf(ucontext_t, "mcontext") + @offsetOf(mcontext_t, "fpregs")),
+          [regspace_offset] "i" (@offsetOf(ucontext_t, "regspace")),
           [sigaltstack] "i" (@intFromEnum(linux.SYS.sigaltstack)),
-          [stack_offset] "p" (@offsetOf(ucontext_t, "stack")),
+          [stack_offset] "i" (@offsetOf(ucontext_t, "stack")),
           [sigprocmask] "i" (@intFromEnum(linux.SYS.rt_sigprocmask)),
-          [sigmask_offset] "p" (@offsetOf(ucontext_t, "sigmask")),
+          [sigmask_offset] "i" (@offsetOf(ucontext_t, "sigmask")),
           [sigset_size] "i" (linux.NSIG / 8),
-        : "memory", "eax", "ecx", "edx"
+        : "cc", "memory", "eax", "ecx", "edx"
     );
 }
 
 pub inline fn getcontext(context: *ucontext_t) usize {
     // This method is used so that getContextInternal can control
     // its prologue in order to read ESP from a constant offset.
     // The unused &getContextInternal input is required so the function is included in the binary.
+    var clobber_edx: usize = undefined;
     return asm volatile (
-        \\ call os.linux.x86.getContextInternal
-        : [ret] "={eax}" (-> usize),
-        : [context] "{edx}" (context),
+        \\ calll %[getContextInternal:P]
+        : [_] "={eax}" (-> usize),
+          [_] "={edx}" (clobber_edx),
```

Comment:
Why is `clobber_edx` required?

Reply:
Inputs are assumed to have the same value unless also an output.  Inputs and outputs can't be clobbered because it can't check that inputs are used before they are clobbered and outputs are implicitly clobbered.